### PR TITLE
Updates code to be compatible with new lints

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Updates codebase to not error on the latest batch of vscode lints provided by the langserver extension. I will bump CI dreamchecker when a version of it comes out.

I have tested map saving, emote cooldowns and overlay compiling. Event stuff has *not* been tested but nothing should break. 

## Why It's Good For The Game
People shouldnt have false positives in the error list.

## Changelog
:cl: AffectedArc07
fix: Improved some code quality
/:cl:

### CODE REVIEW

File: code/modules/awaymissions/maploader/writer.dm [hunk 1]
```diff
@@ -81,22 +81,27 @@
...
 			key_map += "\n"
 
 		key_map += "\n(1,1,[++z_level]) = {\"\n"
+
 		var/z_block = copytext(template_buffer_text, z_pos, findtext(template_buffer_text, ".", z_pos))
-		for(var/y_pos = 1; TRUE; y_pos = findtext(z_block, ";", y_pos) + 1)
+		var/y_pos = 1
+		while(TRUE)
 			if(y_pos >= length(z_block))
 				break
```
Comment (farie82) [inline code/modules/awaymissions/maploader/writer.dm:98]:
```suggestion
		for(var/y_pos = 1; y_pos < length(z_block); z_pos = findtext(z_block, ";", y_pos) + 1)
```

File: code/modules/awaymissions/maploader/writer.dm [hunk 2]
```diff
@@ -81,22 +81,27 @@
...
 
 			var/y_block = copytext(z_block, y_pos, findtext(z_block, ";", y_pos))
 			// A row of keys
-			for(var/x_pos = 1; TRUE; x_pos = findtext(y_block, ",", x_pos) + 1)
+			y_pos = findtext(z_block, ";", y_pos) + 1
+			var/x_pos = 1
+			while(TRUE)
 				if(x_pos >= length(y_block))
 					break
```
Comment (farie82) [inline code/modules/awaymissions/maploader/writer.dm:106]:
```suggestion
			for(var/x_pos = 1; x_pos < length(y_block); x_pos = findtext(y_block, ",", x_pos) + 1)
```

File: code/modules/awaymissions/maploader/writer.dm [hunk 3]
```diff
@@ -105,8 +110,10 @@
 				var/temp_key = keys[key_number]
 				key_map += temp_key
 				CHECK_TICK
+				x_pos = findtext(y_block, ",", x_pos) + 1
```
Comment (farie82) [inline code/modules/awaymissions/maploader/writer.dm:113]:
```suggestion
```

File: code/modules/awaymissions/maploader/writer.dm [hunk 4]
```diff
@@ -105,8 +110,10 @@
...
 			key_map += "\n"
 		key_map += "\"}"
+		z_pos = findtext(template_buffer_text, ".", z_pos) + 1
```
Comment (farie82) [inline code/modules/awaymissions/maploader/writer.dm:116]:
```suggestion
```

File: code/modules/awaymissions/maploader/writer.dm [hunk 5]
```diff
@@ -81,22 +81,27 @@
 	log_debug("Writing out key map...")
 
 	var/list/key_map = list()
-	for(var/z_pos = 1; TRUE; z_pos = findtext(template_buffer_text, ".", z_pos) + 1)
+	var/z_pos = 1
+	while(TRUE)
 		if(z_pos >= length(template_buffer_text))
 			break
```
Comment (farie82) [inline code/modules/awaymissions/maploader/writer.dm:87]:
```suggestion
	for(var/z_pos = 1; z_pos < length(template_buffer_text); z_pos = findtext(template_buffer_text, ".", z_pos) + 1)
```
Can all be made in one for loop

**Review by farie82 — CHANGES_REQUESTED** [review]

**Review by variableundefined — APPROVED** [review]

**Pull request merged**